kong/pdk/vault.lua

```diff
@@ -1455,7 +1455,9 @@ local function new(self)
 
     initialized = true
 
-    if self.configuration.database ~= "off" then
+    if self.configuration.database ~= "off" or  -- postgres
+       self.configuration.role == "data_plane" and self.configuration.cluster_incremental_sync -- incremental dp
```

Comment:
Can you explain why this is bound to incremental sync?

Reply:
It means vault will check the dao events for cache invalidation.